maint: add dev notes and update smoke test

JamieDanielson opened the pull request:
## Which problem is this PR solving?

- my desire to add some notes after my first pass through as someone new to the project

## Short description of the changes

* Add dedicated `DEVELOPING.md` for local dev setup and notes, and reference that section in the README. Also update links in README for other docs.
* Sometimes the console output takes longer than the default timeout of 4000ms, so update Cypress config to timeout at 10000ms instead.

## How to verify that this has the expected result

Following any of the steps in the DEVELOPING doc work, including the smoke tests.

JamieDanielson committed:
maint: add dev notes and update smoke test

* Add dedicated DEVELOPING.md for local dev
setup and notes, and reference that section in
the README. Also update links in README.
* Console output is needed for smoke test, but
DiagConsoleLogger doesn't seem to be logging as
expected, maybe because of the bundler? So for
smoke tests right now, use ConsoleSpanExporter.
* Sometimes the console output takes longer than
the default timeout of 4000ms, so update Cypress
config to timeout at 10000ms instead.

mjayaram reviewed (approved):
All instructions so far worked for me locally! Thanks

ahrbnsn reviewed (approved):
glorious! 🚀

JamieDanielson committed:
oh im silly, diag works

just needed to enable Verbose level
in the console. Removing the
console span exporter

JamieDanielson committed:
update to use run dev, 26 is merged!
